lib/winapi/fileio.c

```diff
@@ -287,6 +287,32 @@ BOOL SetFilePointerEx (HANDLE hFile, LARGE_INTEGER liDistanceToMove, PLARGE_INTE
     }
 }
 
+BOOL SetEndOfFile (HANDLE hFile)
+{
+    // Get current position in file
+    LONG highPosDWORD = 0;
+    PLONG lpHighPosDWORD = &highPosDWORD;
+    DWORD lowPosDWORD = SetFilePointer(hFile, 0, lpHighPosDWORD, FILE_CURRENT);
+
+    // Combine low and high position DWORDs for full position
+    LONGLONG fullCurrentPos;
+    fullCurrentPos = highPosDWORD;
+    fullCurrentPos = fullCurrentPos << 32 | lowPosDWORD;
+
+
+    IO_STATUS_BLOCK ioStatusBlock;
+    FILE_END_OF_FILE_INFORMATION fileEOFInfo;
+    fileEOFInfo.EndOfFile.QuadPart = fullCurrentPos;
+
+    NTSTATUS status = NtSetInformationFile(hFile, &ioStatusBlock, &fileEOFInfo, sizeof(fileEOFInfo), FileEndOfFileInformation);
+    if (NT_SUCCESS(status)) {
+        return FALSE;
```

Comment:
lastError needs to get set here, you can check other functions to see how.

Reply:
I was expecting this, but problem is, I don't know what to set the error to. Is there some kind of standard way we should follow to mimic Windows? Or can we just have our own error messages? Still not sure about the proper way to contribute to WinAPI code.

Reply:
*(for future reviews: If you are aware of missing code / existing issues, please leave a comment in the code or, better yet, add a review comment yourself)*